cclib/parser/xtbparser.py

```diff
@@ -0,0 +1,861 @@
+        """
+        return "Frequency Printout" in line
+
+    def _is_warning(self, line: str) -> bool:
+        """
+        Determine if we are in the warning printout.
+
+        ########################################################################
+        [WARNING] Runtime exception occurred
+        -1- restart_readRestart: Multiplicity missmatch in restart file.
+        ########################################################################
+        """
+        return "[WARNING]" in line
+
+    def _is_hessian_line(self, line: str) -> bool:
+        """
+        Determine if we are in the hessian printout.
+
+        $hessian
+        0.0000000000   0.0000028532   0.0000013880  -0.0000000000  -0.0000008379
+        0.0000004206  -0.0000000000  -0.0000020154  -0.0000018086
+        0.0000028532   0.5449257576  -0.0000002564  -0.0000006835  -0.2724622120
+        0.2128719374  -0.0000021697  -0.2724635456  -0.2128716810
+        0.0000013880  -0.0000002564   0.3976267117   0.0000001891   0.1686647996
+       -0.1988140215  -0.0000015771  -0.1686645432  -0.1988126902
+       -0.0000000000  -0.0000006835   0.0000001891   0.0000000000   0.0000009555
+       -0.0000003616  -0.0000000000  -0.0000002719   0.0000001725
+       -0.0000008379  -0.2724622120   0.1686647996   0.0000009555   0.3096621370
+       -0.1907684616  -0.0000001176  -0.0371999250   0.0221036620
+        0.0000004206   0.2128719374  -0.1988140215  -0.0000003616  -0.1907684616
+        0.1825643728  -0.0000000590  -0.0221034758   0.0162496487
+       -0.0000000000  -0.0000021697  -0.0000015771  -0.0000000000  -0.0000001176
+       -0.0000000590   0.0000000000   0.0000022873   0.0000016361
+       -0.0000020154  -0.2724635456  -0.1686645432  -0.0000002719  -0.0371999250
+       -0.0221034758   0.0000022873   0.3096634705   0.1907680190
+       -0.0000018086  -0.2128716810  -0.1988126902   0.0000001725   0.0221036620
+        0.0162496487   0.0000016361   0.1907680190   0.1825630415
+        """
+        return "$hessian" in line
+
+    def extract(self, inputfile: FileWrapper, line: str) -> None:
+        
+        if self.metadata.get("success") is None:
+            # Initialize as False. Will be overwritten to True if/when appropriate.
+            self.metadata["success"] = False
+
+        version = self._extract_version(line)
+        if version is not None:
+            self.metadata["package_version"] = version
+
+        program_call = self._extract_program_call(line)
+        if program_call is not None:
+            self.metadata["keywords"] = program_call
+
+        coord_filename = self._extract_coord_file(line)
+        if coord_filename is not None:
+            self.metadata["coord_type"] = coord_filename.split(".")[-1]
+
+        method = self._extract_method(line)
+        if method is not None:
+            self.metadata["methods"] = [method]
+
+        charge = self._extract_charge(line)
+        if charge is not None:
+            self.charge = charge
```

Comment:
```suggestion
            self.set_attribute("charge", charge)
```
https://github.com/cclib/cclib/blob/a4831f9c25327f7a49a263da275c9268c58a2f27/cclib/parser/logfileparser.py#L269
Unfortunately there's no equivalent for `metadata` right now.